### PR TITLE
feat: Support existing load balancer backend address pool for agent nodes

### PR DESCRIPTION
**Pull request by xuto2**
**Reason for Change**:
Azure Standard Load Balancer(SLB) introduces some fundamental difference to Azure Basic Load Balancer(BLB). Specifically, vmas/vmss behind Azure SLB don't automatically get internet access like BLB, and it's Azure's recommendation that we "always use outbound rules on a Standard public Load Balancer" (see https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-connections for details).

To support Azure SLB in AKS, AKS needs to maintain the outbound rules of the SLB (and the load balancer rules as well to workaround upstream issue when using SLB https://github.com/kubernetes/kubernetes/pull/76691). So we need the ability to place AKS engine generated nodes (VMSS+VMAS) into an existing SLB's backend address pool.

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
@sylr Does this address any symptoms described in #1099?

**Comment by jackfrancis** [comment]
@xuto2 FYI I rebased on top of master/force-pushed to this branch, so you'll need to pull down for future dev

**Comment by CecileRobertMichon** [comment]
Squash commit message needs to be updated to use semantic message (or if a second commit is pushed the title is enough).

/hold

**Comment by CecileRobertMichon** [comment]
two commits will use PR title as squash message so we're good now.

/hold cancel

File: docs/topics/clusterdefinitions.md [hunk 1]
```diff
@@ -569,7 +569,8 @@ A cluster can have 0 to 12 agent pool profiles. Agent Pool Profiles are used for
...
 | enableVMSSNodePublicIP | no                                                                   | Enable creation of public IP on VMSS nodes. This configuration is only valid on an agent pool with an `"availabilityProfile"` value of `"VirtualMachineScaleSets"`. Defaults to `false`                                                                                                                                                                                                                                                      |
-
+| LoadBalancerBackendAddressPoolIDs | no                                                                   | Enables automatic placement of the agent pool nodes into existing load balancer's backend address pools. Each element value of this string array is the corresponding load balancer backend address pool's Azure Resource Manager(ARM) resource ID.
```
Comment (CecileRobertMichon) [inline docs/topics/clusterdefinitions.md]: Let's add the default value here

Comment (CecileRobertMichon) [reply docs/topics/clusterdefinitions.md]: nevermind, this is a []string

Comment (jackfrancis) [reply docs/topics/clusterdefinitions.md]: added a description of the default case

**Comment by jackfrancis** [comment]
@sylr nevermind I understand what this does, I have another possible fix for the lack of UDP outbound incoming shortly...

**Comment by xuto2** [comment]
Thanks @jackfrancis @CecileRobertMichon @feiskyer !

**Pull request merged**